activerecord/lib/arel/visitors/oracle.rb

```diff
@@ -87,6 +87,48 @@ def visit_Arel_Nodes_Except(o, collector)
           collector << " )"
         end
 
+        def visit_Arel_Nodes_In(o, collector)
+          if Array === o.right && !o.right.empty?
+            o.right.delete_if { |value| unboundable?(value) }
+          end
+
+          if Array === o.right && o.right.empty?
+            collector << "1=0"
+          else
+            first = true
+            o.right.each_slice(in_clause_length) do |sliced_o_right|
+              collector << " OR " unless first
+              first = false
+
+              collector = visit o.left, collector
+              collector << " IN ("
+              visit(sliced_o_right, collector)
+              collector << ")"
+            end
+          end
```

Comment:
I think I missed a `collector` return value here

Reply:
Thanks for the update. Right, I should have been aware of that. I appreciate your kind review and I learned some more visitor patterns.